symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/AuthenticatorFactory.java

```diff
@@ -53,17 +56,28 @@ public AuthenticatorFactory(@Nonnull BdkConfig bdkConfig, @Nonnull ApiClientFact
   public @Nonnull
   BotAuthenticator getBotAuthenticator() throws AuthInitializationException {
     if (this.config.getBot().isCertificateAuthenticationConfigured()) {
+      if (!this.config.getBot().isCertificateConfigurationValid()) {
+        throw new AuthInitializationException(
+            "Only one of certificate path or content should be configured for bot authentication.");
+      }
       return new BotAuthenticatorCertImpl(
           this.apiClientFactory.getSessionAuthClient(),
           this.apiClientFactory.getKeyAuthClient()
       );
     }
-    return new BotAuthenticatorRsaImpl(
-        this.config.getBot().getUsername(),
-        this.loadPrivateKeyFromPath(this.config.getBot().getPrivateKeyPath()),
-        this.apiClientFactory.getLoginClient(),
-        this.apiClientFactory.getRelayClient()
-    );
+    if (this.config.getBot().isRsaAuthenticationConfigured()) {
+      if (!this.config.getBot().isRsaConfigurationValid()) {
+        throw new AuthInitializationException(
+            "Only one of private key path or content should be configured for bot authentication.");
+      }
+      return new BotAuthenticatorRsaImpl(
+          this.config.getBot().getUsername(),
+          this.loadPrivateKeyFromAuthenticationConfig(this.config.getBot()),
+          this.apiClientFactory.getLoginClient(),
+          this.apiClientFactory.getRelayClient()
+      );
+    }
+    throw new AuthInitializationException("Neither rsa private key nor certificate is configured");
```

Comment:
`Neither RSA private key nor certificate is configured.` (RSA in uppercase + dot at the end)